src/misc/api_trace.cc

```diff
@@ -0,0 +1,418 @@
+                                        &ncclRecv_impl,
+                                        &ncclRedOpCreatePreMulSum_impl,
+                                        &ncclRedOpDestroy_impl,
+                                        &ncclGroupStart_impl,
+                                        &ncclGroupEnd_impl,
+                                        &ncclGetVersion_impl,
+                                        &ncclGetUniqueId_impl,
+                                        &ncclCommInitRank_impl,
+                                        &ncclCommInitAll_impl,
+                                        &ncclCommInitRankConfig_impl,
+                                        &ncclCommFinalize_impl,
+                                        &ncclCommDestroy_impl,
+                                        &ncclCommAbort_impl,
+                                        &ncclCommSplit_impl,
+                                        &ncclGetErrorString_impl,
+                                        &ncclGetLastError_impl,
+                                        &ncclCommGetAsyncError_impl,
+                                        &ncclCommCount_impl,
+                                        &ncclCommCuDevice_impl,
+                                        &ncclCommUserRank_impl,
+                                        &ncclMemAlloc_impl,
+                                        &mscclLoadAlgo_impl,
+                                        &mscclRunAlgo_impl,
+                                        &mscclUnloadAlgo_impl,
+                                        &ncclCommRegister_impl,
+                                        &ncclCommDeregister_impl};
+
+        #if defined(RCCL_ROCPROFILER_REGISTER) && RCCL_ROCPROFILER_REGISTER > 0
+            std::array<void*, 1> table_array{&tbl};
+            rocprofiler_register_library_indentifier_t lib_id      = rocprofiler_register_library_indentifier_t{};
+            rocprofiler_register_error_code_t rocp_reg_status =
+                rocprofiler_register_library_api_table("rccl",
+                                                    &ROCPROFILER_REGISTER_IMPORT_FUNC(rccl),
+                                                    ROCP_REG_VERSION,
+                                                    table_array.data(),
+                                                    table_array.size(),
+                                                    &lib_id);
+
+            // INFO( NCCL_COLL,"[rocprofiler-sdk-rccl][ = %d ] rocprofiler-register returned code = %d : %s", getpid(),
+            //     rocp_reg_status, rocprofiler_register_error_string(rocp_reg_status));
+
+            // if(rocp_reg_status != ROCP_REG_SUCCESS && rocp_reg_status != ROCP_REG_NO_TOOLS)
+            //     WARN( "[rocprofiler-sdk-rccl][%d] rocprofiler-register failed with error code %d : %s",  getpid(),
+            //     rocp_reg_status, rocprofiler_register_error_string(rocp_reg_status));
+        #endif
+
+        return &tbl;
```

Comment:
```suggestion
        return tbl;
```
Due to modifications above, `tbl` is now a pointer